client.go

```diff
@@ -85,12 +85,48 @@ func Connect(config ClientConfig) (c *APIClient, err error) {
 		}
 
 		client.Service = service
-		client.auth = auth
+		client.Auth = auth
 	}
 
 	return client, err
 }
 
+// NewAPIClient just returns a new APIClient, without a connection,
+// note: APIClient.Auth, APIClient.Service are left to be set by the caller
+func NewAPIClient(endpoint string, insecure bool, httpClient *http.Client) (c *APIClient, err error) {
```

Comment:
I think if we do need to have a way to pass in an existing http.Client, it would be better to just add a field to the ClientConfig struct where a user can provide it there and use the standard Connect(config) call rather than a variation like this.

Reply:
That sounds right, I'll close this PR and open another one based on your suggestion.